bake/bake.go

```diff
@@ -421,15 +422,18 @@ func (c Config) target(name string, visited map[string]struct{}, overrides map[s
 	}
 	tt := &Target{}
 	for _, name := range t.Inherits {
-		t, err := c.target(name, visited, overrides)
+		if _, ok := inherited[name]; ok {
+			continue
+		}
+		inherited[name] = struct{}{}
+		t, err := c.target(name, visited, inherited, overrides)
```

Comment:
what case is it where passing `inherited` makes sense in here?

Wondering something like:
```
a: foo=123, bar=234
b: inherits:a, bar=567
c: inherits:a, baz=890
d: inherits:b,c
```

So what is the expected value for `bar`. I think it should be `234` as `c` is inherited after `b` and should overwrite its properties.

Reply:
this was in my initial commit and forgot to remove this logic it afterwards. should be ok now.

Reply:
> So what is the expected value for `d->bar`. I think it should be `234` as `c` is inherited after `b` and should overwrite its properties.

And yes should be `234`. I will add another test for this one.